### ISSUE TITLE
PyPI wheels

### ISSUE THREAD

**Issue by boostasbjorn** [estnltk/estnltk#112]
Hi!

Thanks a lot for your hard and ongoing work on this project.

In my company we're using your new 1.6.* version. It works splendidly, and currently we're looking into building estnltk wheels in order to speed up build time.

Is there any way we can help getting more wheels built? We have good experience with Github Actions for example, would you like help to set up automatic wheel building?

**Comment by soras** [comment]
Thank you for the interest and initiative! Great to hear you are using estnltk :)

We are currently using Anaconda packages as the main way of distributing releases, and we haven't automated PyPI wheel building process (we only make 1 wheel per release specifically targeting the Google Colab environment).

Our conda packages currently cover three platforms ( Linux, macOS and Windows ) and Python versions 3.6 - 3.8. 
I think automating PyPI wheel building would make most sense if we could offer the same (or even better?) coverage with the wheels. 
I am not familiar with Github Actions -- how well does it support building for Windows and macOS ? The complex part is building the vabamorf extension, which requires swig and compiling C/C++ code. 

But overall the help on setting up automatic wheel building is welcome :)

**Comment by boostasbjorn** [comment]
Thanks for a quick and thorough response!

I was thinking about using https://cibuildwheel.readthedocs.io/en/stable/. It seems to support almost anything that can be supported (incl. Linux, macOS, Windows).

I'll fork the repo and start working on a GitHub Actions file :grinning:

**Comment by boostasbjorn** [comment]
Here's a PR: #113 

This is what it built in my dev repo: :heavy_check_mark: 
```
$ tree .
.
└── dist
    ├── estnltk-1.6.9b0-cp310-cp310-macosx_10_9_x86_64.whl
    ├── estnltk-1.6.9b0-cp310-cp310-manylinux_2_5_i686.manylinux1_i686.manylinux_2_12_i686.manylinux2010_i686.whl
    ├── estnltk-1.6.9b0-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl
    ├── estnltk-1.6.9b0-cp310-cp310-win32.whl
    ├── estnltk-1.6.9b0-cp310-cp310-win_amd64.whl
    ├── estnltk-1.6.9b0-cp36-cp36m-macosx_10_9_x86_64.whl
    ├── estnltk-1.6.9b0-cp36-cp36m-manylinux_2_5_i686.manylinux1_i686.manylinux_2_12_i686.manylinux2010_i686.whl
    ├── estnltk-1.6.9b0-cp36-cp36m-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl
    ├── estnltk-1.6.9b0-cp36-cp36m-win32.whl
    ├── estnltk-1.6.9b0-cp36-cp36m-win_amd64.whl
    ├── estnltk-1.6.9b0-cp37-cp37m-macosx_10_9_x86_64.whl
    ├── estnltk-1.6.9b0-cp37-cp37m-manylinux_2_5_i686.manylinux1_i686.manylinux_2_12_i686.manylinux2010_i686.whl
    ├── estnltk-1.6.9b0-cp37-cp37m-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl
    ├── estnltk-1.6.9b0-cp37-cp37m-win32.whl
    ├── estnltk-1.6.9b0-cp37-cp37m-win_amd64.whl
    ├── estnltk-1.6.9b0-cp38-cp38-macosx_10_9_x86_64.whl
    ├── estnltk-1.6.9b0-cp38-cp38-manylinux_2_5_i686.manylinux1_i686.manylinux_2_12_i686.manylinux2010_i686.whl
    ├── estnltk-1.6.9b0-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl
    ├── estnltk-1.6.9b0-cp38-cp38-win32.whl
    ├── estnltk-1.6.9b0-cp38-cp38-win_amd64.whl
    ├── estnltk-1.6.9b0-cp39-cp39-macosx_10_9_x86_64.whl
    ├── estnltk-1.6.9b0-cp39-cp39-manylinux_2_5_i686.manylinux1_i686.manylinux_2_12_i686.manylinux2010_i686.whl
    ├── estnltk-1.6.9b0-cp39-cp39-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl
    ├── estnltk-1.6.9b0-cp39-cp39-win32.whl
    ├── estnltk-1.6.9b0-cp39-cp39-win_amd64.whl
    └── estnltk-1.6.9b0.tar.gz
```

**Comment by soras** [comment]
Thank you! The outcome / coverage looks fantastic! :)

One issue, though. I tested some of the created wheels, and it appears they don't work. The installation is OK, but importing  fails due to missing code in the vabamorf extension.
It's a known problem and usually caused by the wrong build order: the vabamorf extension should be the first thing to be built, before anything else in the wheel is pre-compiled or copied to the 'build' folder. Is it possible to force `python setup.py build_ext` before building a wheel ( [this example](https://github.com/estnltk/estnltk/blob/devel_1.6/dev_documentation/PyPi_package_for_Google_Colab/Dockerfile#L34-L37) shows the correct building order )?

**Comment by boostasbjorn** [comment]
Thanks for testing already! Really appreciate the momentum here.

Adding `python setup.py build_ext` is no problem, I have updated the PR :+1: #113

It seems to have resolved the issue according to this little test:
```
$ docker run -it --rm -v $PWD:/data:ro --entrypoint=/bin/bash python:3.8
root@5f2ba5ae1b0b:/# pip install /data/estnltk-1.6.9b0-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl
...
root@5f2ba5ae1b0b:/# python
Python 3.8.12 (default, Sep  3 2021, 20:28:07)
[GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import estnltk
>>>
```
Would you like to test again?

**Comment by soras** [comment]
I tested also the Windows wheel and now it works fine. Splendid! 

Ok, another issue I discovered is that the source distribution does not contain resource and model files. But updating the manifest file should solve that, I see what I can do about it.

**Comment by boostasbjorn** [comment]
Great :ok_hand: 

Alright. Just ping me if you want input on anything.

**Comment by soras** [comment]
PyPI wheels are now available under the estnltk version [1.6.9.1b0](https://pypi.org/project/estnltk/1.6.9.1b0/). They cover Python versions 3.6 - 3.9 on platforms windows-64, linux-64 and macosx_64.

**Comment by boostasbjorn** [comment]
Thank you!

These are really good news, we're already making use of the wheels. We now have a 23% faster build.
